runtime/compiler/env/j9method.cpp

```diff
@@ -6766,30 +6775,52 @@ TR_ResolvedJ9Method::getResolvedDynamicMethod(TR::Compilation * comp, I_32 callS
    // the CP entry is resolved, even in rtResolve mode.
 
    // See if the constant pool entry is already resolved or not
+
+   J9Class    *ramClass = constantPoolHdr();
+   J9ROMClass *romClass = romClassPtr();
+
+   if (unresolvedInCP)
       {
-      TR::VMAccessCriticalSection getResolvedDynamicMethod(fej9());
+      // "unresolvedInCP" is a bit of a misnomer here, but we can describe
+      // something equivalent by checking the callSites table.
+      //
+      *unresolvedInCP = isUnresolvedCallSiteTableEntry(callSiteIndex);
+      }
 
-      J9Class    *ramClass = constantPoolHdr();
-      J9ROMClass *romClass = romClassPtr();
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+   J9InvokeCacheEntry *invokeCache = (J9InvokeCacheEntry *) callSiteTableEntryAddress(callSiteIndex);
```

Comment:
There are a lot of duplication of this code. Can we add a few APIs to get address of member name/appendix and hide the implementation detail of the side table? Something like
```
memberNameAddressFromInvokeDynamicSideTable(int32_t callSiteIndex)
appendixAddressFromInvokeDynamicSideTable(int32_t callSiteIndex)
memberNameAddressFromInvokeHandleSideTable(int32_t cpIndex)
appendixAddressFromInvokeHandleSideTable(int32_t cpIndex)
```

Reply:
Done.